### PR TITLE
Show deleted comments in conversation history

### PR DESCRIPTION
**Pull request by aswin-s**
### Details

This PR fixes an issue where deleted comments get removed completely from conversations. Instead it replaces deleted messages with a placeholder text `[Comment deleted]`

### Fixed Issues

$ https://github.com/Expensify/App/issues/6759

### Tests

1. Send a message in any conversation window.
2. Delete the comment
3. Verify that message gets replaced with placeholder text `[Comment deleted]`
4. Repeat steps 2 & 3 for any previous messages.
5.  Try to delete attachments from web and verify that it gets replaced with placeholder text. 

NB: Native apps doesn't seem to offer delete option at the moment. Reported [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1639767872378300)

### QA Steps

1. Send a message in any conversation window.
2. Delete the comment
3. Verify that message gets replaced with placeholder text `[Comment deleted]`
4. Repeat steps 2 & 3 for any previous messages.
5. Verify that status gets updated at receiving end as well

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

https://user-images.githubusercontent.com/6880914/146600381-84f52c04-cc6e-4c5a-894f-39e86aa1e3c6.mp4

#### Android

https://user-images.githubusercontent.com/6880914/146600404-1df2faf7-908a-4d8d-9d12-cb1a6d5a502c.mp4

### CODE REVIEW

File: src/pages/home/report/ReportActionItem.js [hunk 1]
```diff
@@ -87,15 +87,24 @@ class ReportActionItem extends Component {
...
     }
 
+    /**
+     * Check if the message is deleted
+     * @returns {boolean}
+     */
+    isDeleted() {
+        // A deleted comment has either an empty array or an object with html field with empty string as value
+        return lodashGet(this.props.action, 'message[0].html', '') === '';
+    }
```
Comment (parasharrajat) [inline src/pages/home/report/ReportActionItem.js:97]: Move this to ReportUtils with `isDeletedAction`

File: src/pages/home/report/ReportActionItem.js [hunk 2]
```diff
@@ -87,15 +87,24 @@ class ReportActionItem extends Component {
         this.textInput.focus();
     }
 
+    /**
+     * Check if the message is deleted
+     * @returns {boolean}
+     */
+    isDeleted() {
+        // A deleted comment has either an empty array or an object with html field with empty string as value
+        return lodashGet(this.props.action, 'message[0].html', '') === '';
```
Comment (parasharrajat) [inline src/pages/home/report/ReportActionItem.js:96]: Don't think we need to explicitly compare it with `===' '`

File: src/pages/home/report/ReportActionItemMessage.js [hunk 3]
```diff
@@ -24,16 +28,21 @@ const defaultProps = {
...
-            {_.map(_.compact(props.action.message), (fragment, index) => (
-                <ReportActionItemFragment
-                    key={`actionFragment-${props.action.sequenceNumber}-${index}`}
-                    fragment={fragment}
-                    isAttachment={props.action.isAttachment}
-                    loading={props.action.loading}
-                />
-            ))}
+            {isDeleted
+                ? <ExpensifyText color={themeColors.textSupporting}>{`[${Localize.translateLocal('common.deletedCommentMessage')}]`}</ExpensifyText>
```
Comment (parasharrajat) [inline src/pages/home/report/ReportActionItemMessage.js:37]: use `WithLocalize` instead.

File: src/pages/home/report/ReportActionsView.js [hunk 4]
```diff
@@ -317,14 +317,8 @@ class ReportActionsView extends React.Component {
...
-            .filter((action) => {
-                // Only show non-empty ADDCOMMENT actions or IOU actions
-                // Empty ADDCOMMENT actions typically mean they have been deleted and should not be shown
-                const message = _.first(lodashGet(action, 'message', null));
-                const html = lodashGet(message, 'html', '');
-                return action.actionName === CONST.REPORT.ACTIONS.TYPE.IOU
-                    || (action.actionName === CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT && html !== '');
-            })
+            .filter(action => action.actionName === CONST.REPORT.ACTIONS.TYPE.IOU
+                    || (action.actionName === CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT))
```
Comment (parasharrajat) [inline src/pages/home/report/ReportActionsView.js:321]:
```suggestion
                    || action.actionName === CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT)
```

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Does not look like the supporting color is applied at all? I think you should use styles instead of passing the color.

File: src/pages/home/report/ReportActionItemMessage.js (hunk 3 above)
Comment (aswin-s) [reply src/pages/home/report/ReportActionItemMessage.js]: Done

File: src/pages/home/report/ReportActionItem.js (hunk 1 above)
Comment (aswin-s) [reply src/pages/home/report/ReportActionItem.js]: Done

File: src/pages/home/report/ReportActionItem.js (hunk 2 above)
Comment (aswin-s) [reply src/pages/home/report/ReportActionItem.js]: Fixed

**Comment by aswin-s** [comment]
> Does not look like the supporting color is applied at all? I think you should use styles instead of passing the color.

I took the video right before changing the text style. It was looking dull and getting confusing with normal comments. Updated video below.

https://user-images.githubusercontent.com/6880914/146605070-0966f2e1-b44a-4837-8032-875a35ec1d5b.mp4

File: src/pages/home/report/ReportActionItemMessage.js [hunk 5]
```diff
@@ -42,4 +54,4 @@ ReportActionItemMessage.propTypes = propTypes;
 ReportActionItemMessage.defaultProps = defaultProps;
 ReportActionItemMessage.displayName = 'ReportActionItemMessage';
 
-export default withNetwork()(ReportActionItemMessage);
+export default compose(withNetwork(), withLocalize)(ReportActionItemMessage);
```
Comment (parasharrajat) [inline src/pages/home/report/ReportActionItemMessage.js:57]: NAB: Could you wrap this to the next lines like the other components?

**Comment by parasharrajat** [comment]
Thanks for the video. I am not a designer guy :pencil2:  so I will request @shawnborton  to review the [video](https://github.com/Expensify/App/pull/6819#issuecomment-997018536) and decide the color and font styles.

**Comment by parasharrajat** [comment]
Which one look better? 
image

**Comment by aswin-s** [comment]
Shouldn't format be the same as that used in LHN? @parasharrajat

**Comment by parasharrajat** [comment]
yup, that's a valid point. Just thought of sharing a different perspective. But I definitely think that we should reduce the text size.

**Comment by shawnborton** [comment]
I actually like what's shown in this comment here: https://github.com/Expensify/App/pull/6819#issuecomment-997018536

File: src/libs/reportUtils.js [hunk 6]
```diff
@@ -200,8 +200,19 @@ function canShowReportRecipientLocalTime(personalDetails, myPersonalDetails, rep
         && moment().tz(currentUserTimezone.selected).utcOffset() !== moment().tz(reportRecipientTimezone.selected).utcOffset();
 }
 
+/**
+ * Check if the comment is deleted
+ * @param {Array<Object>} action
```
Comment (parasharrajat) [inline src/libs/reportUtils.js:205]: Wrong type...

File: src/pages/home/report/ReportActionItemMessage.js (hunk 5 above)
Comment (parasharrajat) [inline src/pages/home/report/ReportActionItemMessage.js:57]: NAB: May be wrap this to the next lines like the other components?

File: src/libs/reportUtils.js [hunk 7]
```diff
@@ -200,8 +200,19 @@ function canShowReportRecipientLocalTime(personalDetails, myPersonalDetails, rep
...
+ * @returns {boolean}
```
Comment (parasharrajat) [inline src/libs/reportUtils.js:206]:
```suggestion
 * @returns {Boolean}
```

**Review by parasharrajat — CHANGES_REQUESTED** [review]
a few changes...

File: src/libs/reportUtils.js (hunk 6 above)
Comment (aswin-s) [reply src/libs/reportUtils.js]: Fixed

File: src/pages/home/report/ReportActionItemMessage.js (hunk 5 above)
Comment (aswin-s) [reply src/pages/home/report/ReportActionItemMessage.js]: Done

File: src/libs/reportUtils.js (hunk 7 above)
Comment (aswin-s) [reply src/libs/reportUtils.js]: Fixed

**Review by parasharrajat — APPROVED** [review]
LGTM. 

cc: @Luke9389

**Review by Luke9389 — APPROVED** [review]
Lookin good! Thanks

**Comment by francoisl** [comment]
This failed because of the `import ExpensifyText from '../../../components/ExpensifyText';` statement - looks like that component has been renamed or removed since the last commit. Can someone follow up with a fix PR please? Thanks!

**Comment by marcaaron** [comment]
@francoisl Fix here https://github.com/Expensify/App/pull/7006

**Comment by marcaaron** [comment]
Guessing this did not make it to staging and so does not need to be CP'd to staging but not sure just saw it was breaking my GH actions in open PRs I'm working on.

**Comment by Jag96** [comment]
The changes weren't deployed to staging, but the staging branch does have the broken code on it so I think we should CP so we can get this fixed on the production branch faster.

**Pull request merged**